test/utils/padder/padder.go

```diff
@@ -197,17 +197,34 @@ func (p *Padder) Pad(timeout time.Duration) error {
 // in order to clean all the padding pod in an easier way
 func (p *Padder) Clean() error {
 	pod := &corev1.Pod{}
-	opts := []client.DeleteAllOfOption{
+	delOpts := []client.DeleteAllOfOption{
 		client.InNamespace(p.namespace),
 		client.MatchingLabels{PadderLabel: ""},
 		client.GracePeriodSeconds(5),
 	}
-	err := p.Client.DeleteAllOf(context.TODO(), pod, opts...)
-	if errors.IsNotFound(err) {
-		err = nil
+	if err := p.Client.DeleteAllOf(context.TODO(), pod, delOpts...); err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(p.namespace),
+		client.MatchingLabels{PadderLabel: ""},
+	}
+	if err := p.Client.List(context.TODO(), podList, listOpts...); err != nil {
+		return err
 	}
+
+	for _, padPod := range podList.Items {
```

Comment:
minor, but important: wait should always be done in parallel unless there is a very strong (and documented) reason not to.